lib/app_profiler/backend.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module AppProfiler
+  class Backend
+    def run(params = {}, &block)
+      raise NotImplementedError
+    end
+
+    def start(params = {})
+      raise NotImplementedError
+    end
+
+    def stop
+      raise NotImplementedError
+    end
+
+    def results
+      raise NotImplementedError
+    end
+
+    def running?
+      raise NotImplementedError
+    end
+  end
+
+  autoload :StackprofBackend, "app_profiler/backend/stackprof"
+  begin
+    require "app_profiler/backend/vernier"
```

Comment:
Typically, we define peer deps by using `gem` like you've done and make the file that needs to autoloaded. We can't expect that every developer will want to use vernier, so raising a warning here doesn't make sense. Let's remove it and just autoload the constant instead. The `gem` line you have in the vernier backend will raise an appropriate error.